src/hotspot/share/classfile/stringTable.cpp

```diff
@@ -796,8 +796,7 @@ void StringTable::allocate_shared_strings_array(TRAPS) {
       // This can only happen if you have an extremely large number of classes that
       // refer to more than 16384 * 16384 = 26M interned strings! Not a practical concern
       // but bail out for safety.
-      log_error(cds)("Too many strings to be archived: " SIZE_FORMAT, _items_count);
-      os::_exit(1);
+      MetaspaceShared::unrecoverable_writing_error(err_msg("Too many strings to be archived: %ld", _items_count));
```

Comment:
Why is log_error() removed and passed to unrecoverable_writing_error?  Seems like this is breaking the rule of simple strings can be passed to unrecoverable_writing_error().  The inconsistency of passing a message vs logging outside the call is very strange to me.